airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab.tsx

```diff
@@ -1,241 +1,171 @@
+            return;
+          }
+        } else {
+          // The catalog hasn't changed. We don't need to ask for any confirmation and can simply save.
+          await saveConnection(formValues, true, catalogHasChanged);
+        }
+
+        setSaved(true);
+        setSchemaHasBeenRefreshed(false);
+      } catch (e) {
+        setSubmitError(e);
       }
-      // Save the connection taking into account the correct skipRefresh value from the dialog choice.
-      await saveConnection(values, { skipReset: !result.reason });
-    } else {
-      // The catalog hasn't changed. We don't need to ask for any confirmation and can simply save.
-      await saveConnection(values, { skipReset: true });
-    }
-  };
-
-  // TODO: Move this into the service next
-  const refreshSourceSchema = async () => {
-    setSaved(false);
-    setActiveUpdatingSchemaMode(true);
-    const { catalogDiff, syncCatalog } = await refreshCatalog();
-    if (catalogDiff?.transforms && catalogDiff.transforms.length > 0) {
-      await openModal<void>({
-        title: formatMessage({ id: "connection.updateSchema.completed" }),
-        preventCancel: true,
-        content: ({ onClose }) => (
-          <CatalogDiffModal catalogDiff={catalogDiff} catalog={syncCatalog} onClose={onClose} />
-        ),
-      });
-    }
-  };
-
-  const onRefreshSourceSchema = async () => {
-    if (formDirty) {
-      // The form is dirty so we show a warning before proceeding.
-      openConfirmationModal({
-        title: "connection.updateSchema.formChanged.title",
-        text: "connection.updateSchema.formChanged.text",
-        submitButtonText: "connection.updateSchema.formChanged.confirm",
-        onSubmit: () => {
-          closeConfirmationModal();
-          refreshSourceSchema();
-        },
-      });
-    } else {
-      // The form is not dirty so we can directly refresh the source schema.
-      refreshSourceSchema();
-    }
-  };
+    },
+    [
+      connection.connectionId,
+      connection.operations,
+      connection.status,
+      connection.syncCatalog.streams,
+      connectionService,
+      formatMessage,
+      mode,
+      openModal,
+      saveConnection,
+      setSchemaHasBeenRefreshed,
+      setSubmitError,
+      workspaceId,
+    ]
+  );
 
-  const onCancelConnectionFormEdit = () => {
-    setSaved(false);
-    setActiveUpdatingSchemaMode(false);
-  };
+  useConfirmCatalogDiff(connection);
 
   return (
     <div className={styles.content}>
-      {!isRefreshingCatalog && connection ? (
-        <ConnectionFormServiceProvider
-          connection={connection}
-          mode={connection?.status !== ConnectionStatus.deprecated ? "edit" : "readonly"}
-          onSubmit={onSubmitForm}
-          onCancel={onCancelConnectionFormEdit}
-          formId={formId}
-          formDirty={formDirty}
+      {schemaError && !schemaRefreshing ? (
+        <SchemaError schemaError={schemaError} />
```

Comment:
Pulled the SchemaError logic over from CreateConnectionForm in case schema refreshing fails.